api/envoy/api/v2/discovery.proto

```diff
@@ -94,6 +95,12 @@ message DiscoveryResponse {
   // DiscoveryRequest bearing the nonce. The nonce is optional and is not
   // required for non-stream based xDS implementations.
   string nonce = 5;
+
+  // Duration for which the resources received as part of this responses are
+  // valid for use. If not set, there is no time limit on use of the resources.
+  // This allows the management server to associate freshness to information
+  // sent to make sure envoys are not working with stale/incorrect information.
+  google.protobuf.Duration resources_lease_duration = 6;
```

Comment:
should we be using these new fields in this PR?

Reply:
+1. I would like an implementation commitment for this feature, since it has substantial implications for management server and Envoy behavior. I know a lot of this work is happening in the context of gRPC-LB, where xDS is being used but the client is not Envoy. 

For this specific feature, I think the Envoy-side behavior shouldn't be hard to add. We need to setup timers and issue invalidated `onConfigUpdate` calls when it expires.

Reply:
Thanks for the comments, I can do the following 
  - propose behavior for each DS (I am not sure it applies to all DS exactly the same way)
  - provide implementation for EDS/CDS,

I will not be able to address it this week as I am traveling. I can keep this PR open and update it or open a new one later.